core/common/src/main/java/alluxio/underfs/Fingerprint.java

```diff
@@ -135,6 +140,40 @@ public String serialize() {
     return sb.toString();
   }
 
+  /**
+   * Returns true if the serialized fingerprint matches the fingerprint in metadata.
+   *
+   * @param serializedFP a finger print in string representation
+   *
+   * @return true if the serialized fingerprint matches the current fingerprint in metadata
+   */
+  public boolean matchMetadata(String serializedFP) {
```

Comment:
consider changing this API to match against other `FingerPrint` instances. It would be more symmetric and save a parse if the caller wants to call both `matchMetadata` and `matchContent`